sklearn/cluster/optics_.py

```diff
@@ -619,12 +619,12 @@ def cluster_optics_xi(reachability, predecessor, ordering, min_samples,
     n_samples = len(reachability)
     _validate_size(min_samples, n_samples, 'min_samples')
     if min_samples <= 1:
-        min_samples = max(2, min_samples * n_samples)
+        min_samples = int(round(max(2, min_samples * n_samples)))
```

Comment:
I'll prefer to use `int` on the part that needs to be converted to an int:

```suggestion
        min_samples = max(2, int(min_samples * n_samples))
```